### PR TITLE
Add sponsor

### PR DESCRIPTION
**Pull request by railsstudent**
This is static page.
I still have no clue on the integration of react with jinja and read data from sponsor.yaml.

Any feedback on the css and layout is welcome.

### CODE REVIEW

**Review by yookoala — COMMENTED** [review]
Recommend to use YAML's folder style for multi-line string. Much easier to read and maintain. Also, quotes (") are optional for string values.

Reference: 
* [Stack Overflow - In YAML, how do I break a string over multiple lines?](http://stackoverflow.com/questions/3790454/in-yaml-how-do-i-break-a-string-over-multiple-lines)
* [YAML spec 1.2](http://www.yaml.org/spec/1.2/spec.html#id2760844)

**Review by yookoala — APPROVED** [review]

File: assets/data/sponsor.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+           - "and Web developers."
+        logos:
+          - title: JetBrains
+            alt: "JetBrains logo"
+            img: "//2016.opensource.hk/statics/images/sponsors/logo_jetbrains_3_500.jpg"
+            url: "https://www.jetbrains.com/"
+        links:
+          - text: "JetBrains Website"
+            url: "https://www.jetbrains.com/"
+patrons:
```
Comment (tonyhhyip) [inline assets/data/sponsor.yml]: Remove patrons before we get some name

File: assets/pages/sponsor/index.jinja [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+          <p>We would like to thank the following sponsors.</p>
+
+          <div class="sponsor-types">
+            <h3>Gold</h3>
+            <div class="sponsor-type">
+              <div class="row">
+                <div class="col s3">
+                  <p class="logo">
+                    <a href="http://gandi.net" title="Gandi.net" target="_blank">
+                      <img src="//2016.opensource.hk/statics/images/sponsors/gandi_logo_black.jpg" alt="Gandi.net logo">
```
Comment (tonyhhyip) [inline assets/pages/sponsor/index.jinja]: Replace with `//file.hkoscon.org/sponsor/gandi.jpg`

File: assets/pages/sponsor/index.jinja [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                </div>
+              </div>
+            </div> <!-- gold -->
+            <h3>Prize Sponsor</h3>
+            <div class="sponsor-type">
+              <div class="row">
+                <div class="col s3">
+                  <p class="logo">
+                    <a href="https://www.jetbrains.com/" title="JetBrains" target="_blank">
+                      <img src="//2016.opensource.hk/statics/images/sponsors/logo_jetbrains_3_500.jpg" alt="JetBrains logo">
```
Comment (tonyhhyip) [inline assets/pages/sponsor/index.jinja]: Replace with `//file.hkoscon.org/sponsor/jetbrains.jpg`

File: assets/pages/sponsor/index.jinja [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+                    <div>
+                      <a class="waves-effect waves-light btn" target="_blank"
+                      href="https://www.jetbrains.com/" role="button">JetBrains Website</a>
+                    </div>
+                </div>
+              </div>
+            </div> <!-- general -->
+          </div> <!-- sponsor-types -->
+
+          <!-- Patron -->
```
Comment (tonyhhyip) [inline assets/pages/sponsor/index.jinja]: Use `{#-` and `#}` instead to reduce render output size

File: assets/data/sponsor.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+          - "provider. Created in France in 1999, Gandi is now present in 3 "
+          - "continents,with offices in Paris, San Francisco, Luxembourg, and "
+          - "Taipei. We position ourselves as the next port of call for those "
+          - "who want to create an online presence, right after Internet "
+          - "access providers, since domain names are the firststep in "
+          - "creating your own piece of the Internet."
+        logos:
+          - title: "Gandi.net"
+            alt: "Gandi.net logo"
+            img: "//2016.opensource.hk/statics/images/sponsors/gandi_logo_black.jpg"
```
Comment (tonyhhyip) [inline assets/data/sponsor.yml]: Replace with `//file.hkoscon.org/sponsor/gandi.jpg`

File: assets/data/sponsor.yml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+           - "Professional Developers providing a complete set of tools for "
+           - "productive Python, Web and Scientific development. PyCharm comes "
+           - "in two editions. The free and open-source PyCharm Community "
+           - "Edition is perfect for pure Python and scientific development. "
+           - "PyCharm Professional Edition is designed for professional Python "
+           - "and Web developers."
+        logos:
+          - title: JetBrains
+            alt: "JetBrains logo"
+            img: "//2016.opensource.hk/statics/images/sponsors/logo_jetbrains_3_500.jpg"
```
Comment (tonyhhyip) [inline assets/data/sponsor.yml]: Replace with //file.hkoscon.org/sponsor/jetbrains.jpg

**Review by tonyhhyip — CHANGES_REQUESTED** [review]
Some link is not correct

**Comment by yookoala** [comment]
A quick note about the yaml data. All data in the yaml files of data folder are loaded into a single `data` object by the gulp script (through the `requireyml` library and nunjuck).

For example, [assets/data/site.yml](assets/data/site.yml) loaded as `data.site`.

File: assets/data/sponsor.yml (hunk 5 above)
Comment (railsstudent) [reply assets/data/sponsor.yml]: Fixed

**Review by tonyhhyip — APPROVED** [review]

**Pull request merged**